### ISSUE TITLE
Batch Norm Explained Visually

### ISSUE THREAD

**Issue by chippen001** [chippen001/Capstone-Resources#4]
### Title
Batch Norm Explained Visually -- How it works, and why neural networks need it

### URL
https://towardsdatascience.com/batch-norm-explained-visually-how-it-works-and-why-neural-networks-need-it-b18919692739 

### Summary 
This article describes batch normalization works in deep learning. Standardization, scale and shift factors, and the process of training mini-batches. 

### Key Points 

- Batch Norm is neural network layer. Often added as part of Linear or Convolutional block to stabilize network during training.
- Standard practice to normalize data to zero mean and unit variance when inputting it into a deep learning model. This puts features on the same scale.
- Batch Norm works as another network layer put between hidden layers to normalize outputs and inputs passing between layers. 
- 2 learnable parameters called beta and gamma. 2 non-learnable parameters Mean Moving Average and Variance Moving Average. Parameters are used in each Batch Norm layer. 
- Training of mini-batches:

1. Activations from previous layer passed as input to Batch Norm.
2. Mean and variance of values in mini-batch calculator for each activation vector.
3. Normalized values calculated for each activation feature vector using mean/variance.
4. Normalized values multiplied by gamma and add beta. These factors are trainable and can learn, shift and scale, and get better predictions.
5. Hyperparameter Moving Average uses scalar 'momentum'.

### Citation
Ketan Doshi. Batch norm explained visually – how it works, and why neural networks need it. Medium, 2021.
